mmdet/structures/bbox/base_bbox.py

```diff
@@ -0,0 +1,412 @@
+        pass
+
+    @abstractproperty
+    def heights(self) -> Tensor:
+        """Return a tensor representing the heights of boxes."""
+        pass
+
+    @abstractclassmethod
+    def flip(self: T,
+             img_shape: Tuple[int, int],
+             direction: str = 'horizontal') -> T:
+        """Flip bboxes horizontally or vertically.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+            direction (str): Flip direction, options are "horizontal",
+                "vertical" and "diagonal". Defaults to "horizontal"
+
+        Returns:
+            T: Flipped boxes.
+        """
+        pass
+
+    @abstractclassmethod
+    def translate(self: T, distances: Tuple[float, float]) -> T:
+        """Translate bboxes.
+
+        Args:
+            distances (Tuple[float, float]): translate distances. The first
+                is horizontal distance and the second is vertical distance.
+
+        Returns:
+            T: Translated boxes.
+        """
+        pass
+
+    @abstractclassmethod
+    def clip(self: T, img_shape: Tuple[int, int]) -> T:
+        """Clip boxes according to border.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+
+        Returns:
+            T: Cliped boxes.
+        """
+        pass
+
+    @abstractclassmethod
+    def rotate(self: T,
+               center: Tuple[float, float],
+               angle: float,
+               img_shape: Optional[Tuple[int, int]] = None) -> T:
+        """Rotate all boxes.
+
+        Args:
+            center (Tuple[float, float]): Rotation origin.
+            angle (float): Rotation angle.
+            img_shape (Tuple[int, int], Optional): image shape.
+                Defaults to None.
+
+        Returns:
+            T: Rotated boxes.
+        """
+        pass
+
+    @abstractclassmethod
+    def project(self: T,
+                homography_matrix: Union[Tensor, np.ndarray],
+                img_shape: Optional[Tuple[int, int]] = None) -> T:
+        """Geometric transformation for bbox.
+
+        Args:
+            homography_matrix (Tensor or np.ndarray]):
+                Shape (3, 3) for geometric transformation.
+            img_shape (Tuple[int, int], optional): Image shape.
+                Defaults to None.
+
+        Returns:
+            T: Converted bboxes.
+        """
+        pass
+
+    @abstractclassmethod
+    def rescale(self: T, rescale_factor: Tuple[float, float]) -> T:
+        """Rescale boxes w.r.t. rescale_factor.
+
+        Args:
+            rescale_factor (Tuple[float, float]): factors for scaling boxes.
+                The length should be 2.
+
+        Returns:
+            T: Rescaled boxes.
+        """
+        pass
+
+    @abstractclassmethod
+    def rescale_size(self: T, rescale_factor: Tuple[float, float]) -> T:
+        """Only rescale the box shape. The centers of boxes are unchanged.
```

Comment:
Can not directly understand this method will keep the centers unchanged.

Reply:
The `rescale_size` only scales the box width and height, which is different from `rescale` which changes both the centers and shapes.

I will change the function name to a more distinctive one to differentiate from `rescale`.